float32 linear_speed
float32 angular_speed
bool shoot_cannon 